alf/utils/checkpoint_utils.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2020 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import warnings
+import torch
+
+
+class Checkpointer(object):
+    """A checkpoint manager for saving and loading checkpoints."""
+
+    def __init__(self, ckpt_dir, **kwargs):
+        """A class for making checkpoints.
+
+        Args:
+            ckpt_dir: The directory to save checkpoints. Create ckpt_dir if
+                it doesn't exist.
+            kwargs: Items to be included in the checkpoint. Each item needs
+                to have state_dict and load_state_dict implemented.
+        """
+
+        self._modules = kwargs
+        self._ckpt_dir = ckpt_dir
+        self._global_step = -1
+
+        os.makedirs(self._ckpt_dir, exist_ok=True)
+
+    def load(self, global_step="latest"):
+        """Load checkpoint
+        Args:
+            global_step: the number of training steps corresponding to the
+                current state to be saved. It will be appended to the name of the chekpoint as a suffix.
+        Returns:
+            next_step_num (int): the step_number for the next training step
+        """
+
+        def _load_checkpoint(checkpoint):
+            self._global_step = checkpoint["global_step"]
+            for k, v in self._modules.items():
+                self._modules[k].load_state_dict(checkpoint[k])
+
+        f_path_latest = os.path.join(self._ckpt_dir, "latest")
```

Comment:
handle the case if latest does not exist.

Reply:
Line 55 checks for the existence of latest, and if it does not exist, a warning will be given (Line 62).